### PR TITLE
PMM-12702 Fix SSLSni checker.

### PR DESCRIPTION
**Pull request by BupycHuk**
PMM-12702

Link to the Feature Build: SUBMODULES-3449

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

File: managed/models/agentversion.go [hunk 1]
```diff
@@ -37,18 +36,22 @@ func (e *AgentNotSupportedError) Error() string {
 	return fmt.Sprintf("'%s' functionality is not supported by pmm-agent %q version %q. Required minimum version is %q", e.Functionality,
 		e.AgentID, e.AgentVersion, e.MinAgentVersion)
 }
+func (e *AgentNotSupportedError) Is(err error) bool {
+	_, ok := err.(*AgentNotSupportedError)
+	return ok
+}
```
Comment (artemgavrilov) [inline managed/models/agentversion.go:42]: Why it's needed? It's analog of `errors.Is()`, no?

File: managed/models/agentversion.go [hunk 2]
```diff
@@ -17,7 +17,6 @@ package models
 
 import (
 	"fmt"
-
```
Comment (ademidoff) [inline managed/models/agentversion.go:20]: It looks like it needs to get `make format`'ed ❓

File: managed/models/agentversion.go (hunk 1 above)
Comment (BupycHuk) [reply managed/models/agentversion.go:42]: errors.Is didn't work correctly and actually calls this method to compare, so I had to implement it this way.

Comment (BupycHuk) [reply managed/models/agentversion.go:43]: https://github.com/percona/pmm/pull/2625/files#diff-8792e067d067e501754c905b86a46d9101c2861f211cc265cdc5fd395a2e4bc4R81 here is where errors.Is is called

Comment (artemgavrilov) [reply managed/models/agentversion.go:43]: It seems that we need `As` there instead of `Is`. We need to check error type, but not exact copy.

Comment (BupycHuk) [reply managed/models/agentversion.go:43]: As didn't work as well

Comment (artemgavrilov) [reply managed/models/agentversion.go:43]: Why? What's so special about this case?

Comment (BupycHuk) [reply managed/models/agentversion.go:43]: I have no idea

Comment (BupycHuk) [reply managed/models/agentversion.go]: the reason was because `IsAgentSupported` was returning a pointer and that's why it didn't work.

**Review by ademidoff — APPROVED** [review]

**Review by artemgavrilov — APPROVED** [review]

**Pull request merged**